__tests__/models.test.js

```diff
@@ -99,6 +99,133 @@ describe("GET /api/reviews", () => {
+              expect(msg).toBe("Bad Request: Category does not exist!");
+            });
+        });
+        test("200: category exists but has no reviews", () => {
+          return request(app)
+            .get("/api/reviews?category=children's games")
+            .expect(200)
+            .then(({ body: { reviews } }) => {
+              expect(reviews).toEqual([]);
+            });
+        });
+        // autopasses as the code doesn't allow for incorrect categories, thus never needs promise reject at row count 0
+      });
+    });
+    describe("sort_by", () => {
+      test("200: reviews can be sorted by different columns via query", () => {
+        return request(app)
+          .get("/api/reviews?sort_by=title")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSortedBy("title", { descending: true });
+          });
+      });
+      test("200: reviews are sorted by date by default", () => {
+        return request(app)
+          .get("/api/reviews")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSortedBy("created_at", { descending: true });
+          });
+      });
+      describe("Error Handling:", () => {
+        test("400: column name does not exist (42703)", () => {
+          // this test uses psql error code 42703 to throw back an error, potential for sql injection maybe?
+          return request(app)
+            .get("/api/reviews?sort_by=date")
+            .expect(400)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe("Bad Request: Column does not exist!");
+            });
+        });
+        test("400: only accepts set column names", () => {
+          // the method to pass this test uses custom error, with an accepted query values array, which means the sql error is never thrown
+          // I can delete the previous test as this tests for both,  but should I?
+          // I think it is dependent on the code written, so both tests are maybe needed?
+          return request(app)
+            .get("/api/reviews?sort_by=date")
+            .expect(400)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe("Bad Request: Column does not exist!");
+            });
+        });
+      });
+    });
+    describe("order_by", () => {
+      test("200: reviews are able to be sorted in ascending/descending", () => {
+        return request(app)
+          .get("/api/reviews?order_by=asc")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSortedBy("created_at");
+          });
+      });
+      test("200: reviews are in descending order by default", () => {
+        return request(app)
+          .get("/api/reviews")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSortedBy("created_at", { descending: true });
+          });
+      });
```

Comment:
This also does not need to be tested as it is tested in the original get request.